### PR TITLE
Roles PR

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

**Comment by sopel39** [comment]
@martint @electrum please take a look at the PR and commits structure if they are fine with CLA

**Comment by sopel39** [comment]
For future reference this code was initially created by @arhimondr @cawallin @anusudarsan while at TD and massaged a lot by @kokosing, @sopel39, @anusudarsan, @findepi during many PR incarnations to prestodb

**Comment by sopel39** [comment]
Our internal infra passed

**Comment by tooptoop4** [comment]
gentle ping for prestodb merge

File: presto-parser/src/main/java/io/prestosql/sql/parser/AstBuilder.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
```
Comment (wenleix) [inline presto-parser/src/main/java/io/prestosql/sql/parser/AstBuilder.java]: unrelated change
